src/Device/Driver/LX_EOS/LXEosDeclare.cpp

```diff
@@ -0,0 +1,218 @@
+  data.byClass = 0;
+
+  memset(data.observer, 0x00, sizeof(data.observer));
+  memset(data.gps, 0x00, sizeof(data.gps));
+
+  // Number of turnpoints (excluding start and finish)
+  data.num_of_tp = declaration.Size() - 2;
+
+  // Takeoff point (set to all zeroes, as it is not useful)
+  data.prg[0] = 3;
+  data.lat[0] = 0;
+  data.lon[0] = 0;
+  CopyStringSpacePadded(data.name[0], _T("TAKEOFF"), 9);
+
+  for (uint8_t i = 1; i <= declaration.Size(); i++) {
+    data.prg[i] = 1;
+    data.lat[i] = ConvertCoord(declaration.GetLocation(i - 1).latitude);
+    data.lon[i] = ConvertCoord(declaration.GetLocation(i - 1).longitude);
+    CopyStringSpacePadded(data.name[i], declaration.GetName(i - 1), 9);
+  }
+
+  /*
+   * Setting the landing seems to not work.
+   * Eos will either show name of waypoint that used to be on that position in
+   * list before declaration, or "NO_NAME". It is more likely a bug in the Eos
+   * firmware than in the structure of the command. It seems to not cause any
+   * practical issues, so let's ignore it.
+   */
+  data.prg[declaration.Size() + 1] = 2;
+  data.lat[declaration.Size() + 1] = 0;
+  data.lon[declaration.Size() + 1] = 0;
+  CopyStringSpacePadded(data.name[declaration.Size() + 1], _T("LANDING"), 9);
+
+  // Set the remaining entries to zeroes
+  for (uint8_t i = declaration.Size() + 2; i < 12; i++) {
+    data.prg[i] = 0;
+    data.lat[i] = 0;
+    data.lon[i] = 0;
+    memset(data.name[i], 0x00, 9);
+  }
+
+  // CRC is calculated only from data bytes (excluding the SYN and CMD)
+  // Using poly 0x69 and initial 0xFF
+  data.crc = Calculate8bitCRC(
+    reinterpret_cast<uint8_t*>(&data) + 2, sizeof(data) - 3, 0xFF, 0x69);
+  std::span<std::byte> message(reinterpret_cast<std::byte*>(&data),
+                               sizeof(data));
+
+  return WriteAndWaitForACK(message, env);
+}
+
+bool
+LXEosDevice::SendObsZone(uint8_t tp_nr,
+                         const Declaration& declaration,
+                         OperationEnvironment& env)
+{
+  if (tp_nr < 1 || tp_nr > declaration.Size())
+    return false;
+
+  EosObsZoneStruct data;
+  const auto& turnpoint = declaration.turnpoints.at(tp_nr - 1);
+
+  data.uiTpNr = tp_nr;
+  data.uiR1 = ToLE32(turnpoint.radius);
+
+  if (tp_nr == 1)                       // Start
+    data.uiDirection = 2;               // Next
+  else if (tp_nr == declaration.Size()) // Finish
+    data.uiDirection = 3;               // Previous
+  else                                  // Turnpoint
+    data.uiDirection = 0;               // Symmetric
+
+  data.bIsLine = (turnpoint.shape == Declaration::TurnPoint::LINE) ? 1 : 0;
+
+  /* We cannot be sure about the format of float on the platform, hardcoded
+   * bytes are used for the angles */
+
+  // Bytes of float values for 45, 90 and 180 degree angles in radians
+  uint8_t angle180[] = { 0xdb, 0x0f, 0x49, 0x40 };
+  uint8_t angle90[] = { 0xdb, 0x0f, 0xc9, 0x3f };
+  uint8_t angle45[] = { 0xdb, 0x0f, 0x49, 0x3f };
```

Comment:
This is extremely obscure and for no good reason, it allocates 12 bytes on the stack at runtime on every call and initializes them. If you really need constants like that, make them `static constexpr`, but I doubt this is the right way to do it.

Reply:
Can I assume that all platforms use IEEE 754 float? How is float affected by big/little endian?

Reply:
> Can I assume that all platforms use IEEE 754 float?

You need to check https://en.cppreference.com/w/cpp/types/numeric_limits/is_iec559 - I agree to make this a `static_assert`, until we see that a currently supported target doesn't support it.

> How is float affected by big/little endian?

I don't know, I never cared. But if you write code that transmits floats in a binary stream, it's up to you to figure out what byte order (and bit format) is expected/emitted by the device you're talking to, and you must figure out the byte order (and bit format) of all CPUs supported by XCSoar.

As I wrote in another comment, this shouldn't be sprinkled everywhere, but implemented in one central type, like I did with classes like `PackedLE32`.